src/pages/index.js

```diff
@@ -0,0 +1,290 @@
+import tw, { css } from 'twin.macro'
+import Head from 'next/head'
+import { useEffect, useState } from 'react'
+import { createMachine, assign } from 'xstate'
+import { useMachine } from '@xstate/react'
+
+export default function Home() {
+  const currentLineNumber = useCurrentLineNumber()
+  return (
+    <>
+      <Head>
+        <title>Into My Own</title>
+      </Head>
+      <div
+        id="textContainer"
+        tw="flex justify-center my-0 mx-auto min-h-screen sticky top-0"
+      >
+        <div tw="py-16 whitespace-nowrap">
+          <h1
+            css={[
+              tw`font-body font-bold text-2xl sm:text-2xl md:text-4xl lg:text-4xl`,
+              css`
+                animation: from-on-to-past 3s forwards ease-out;
+                color: #ff90d6;
+              `,
+            ]}
+          >
+            Into My Own
+          </h1>
+          {stanzas.map((lines, stanzaIdx) => {
+            const previousStanzas = stanzas.slice(0, stanzaIdx)
+            const previousLineNum = previousStanzas.reduce(
+              (totalLines, linesInStanza) => totalLines + linesInStanza.length,
+              0
+            )
+            return (
+              <Paragraph key={stanzaIdx}>
+                {lines.map((line, lineIdx) => {
+                  const lineNumber = previousLineNum + lineIdx
+                  // events: 'SCROLL_ON' | 'SCROLL_PAST' | 'SCROLL_BEFORE'
+                  const animationEvent =
+                    currentLineNumber === lineNumber
+                      ? 'SCROLL_ON'
+                      : currentLineNumber > lineNumber
+                      ? 'SCROLL_PAST'
+                      : currentLineNumber < lineNumber
+                      ? 'SCROLL_BEFORE'
+                      : null
+                  return (
+                    <Line
+                      key={line}
+                      animationEvent={animationEvent}
+                      lastLine={lineNumber === lastLineNumber}
+                    >
+                      {line}
+                    </Line>
+                  )
+                })}
+                {/* add the attribution to the end of the last stanza */}
+                {stanzaIdx === stanzas.length - 1 ? (
+                  <Attribution
+                    animationEvent={
+                      currentLineNumber === lastLineNumber
+                        ? 'SCROLL_ON'
+                        : 'SCROLL_BEFORE'
+                    }
+                  >
+                    — Robert Frost
+                  </Attribution>
+                ) : null}
+              </Paragraph>
+            )
+          })}
+        </div>
+      </div>
+      <div
+        id="scrollBox"
```

Comment:
You don't have to, but if you want, you can change this to a ref. At the top of the component add this:
```js
const scrollBoxRef = useRef(null)
```

Then pass `scrollBoxRef` into `useCurrentLineNumber`

Then inside the `useEffect`, change the `querySelector` to this:
```js
const scrollBox = scrollBoxRef.current
if (scrollBox === null) return
```